tests/components/kaleidescape/test_media_player.py

```diff
@@ -0,0 +1,187 @@
+"""Tests for Kaleidescape media player platform."""
+
+import asyncio
+from unittest.mock import MagicMock
+
+from kaleidescape import const as kaleidescape_const
+from kaleidescape.device import Movie
+
+from homeassistant.components.media_player.const import DOMAIN as MEDIA_PLAYER_DOMAIN
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_MEDIA_NEXT_TRACK,
+    SERVICE_MEDIA_PAUSE,
+    SERVICE_MEDIA_PLAY,
+    SERVICE_MEDIA_PREVIOUS_TRACK,
+    SERVICE_MEDIA_STOP,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_IDLE,
+    STATE_OFF,
+    STATE_PAUSED,
+    STATE_PLAYING,
+)
+from homeassistant.core import HomeAssistant
+
+from . import MOCK_SERIAL
+
+from tests.common import MockConfigEntry
+
+ENTITY_ID = f"media_player.kaleidescape_device_{MOCK_SERIAL}"
+FRIENDLY_NAME = f"Kaleidescape Device {MOCK_SERIAL}"
+
+
+async def test_entity(
+    hass: HomeAssistant,
+    mock_device: MagicMock,
+    mock_integration: MockConfigEntry,
+) -> None:
+    """Test entity attributes."""
+    entity = hass.states.get(ENTITY_ID)
+    assert entity is not None
+    assert entity.state == STATE_OFF
+    assert entity.attributes["friendly_name"] == FRIENDLY_NAME
+
+
+async def test_update_state(
+    hass: HomeAssistant,
+    mock_device: MagicMock,
+    mock_integration: MockConfigEntry,
+) -> None:
+    """Tests dispatched signals update player."""
+    entity = hass.states.get(ENTITY_ID)
+    assert entity is not None
+    assert entity.state == STATE_OFF
+
+    # Device turns on
+    mock_device.power.state = kaleidescape_const.DEVICE_POWER_STATE_ON
+    mock_device.dispatcher.send(kaleidescape_const.DEVICE_POWER_STATE)
+    await asyncio.sleep(0)
+    await hass.async_block_till_done()
+    entity = hass.states.get(ENTITY_ID)
+    assert entity is not None
+    assert entity.state == STATE_IDLE
+
+    # Devices starts playing
+    mock_device.movie = Movie(
+        handle="handle",
+        title="title",
+        cover="cover",
+        cover_hires="cover_hires",
+        rating="rating",
+        rating_reason="rating_reason",
+        year="year",
+        runtime="runtime",
+        actors=[],
+        director="director",
+        directors=[],
+        genre="genre",
+        genres=[],
+        synopsis="synopsis",
+        color="color",
+        country="country",
+        aspect_ratio="aspect_ratio",
+        media_type="media_type",
+        play_status=kaleidescape_const.PLAY_STATUS_PLAYING,
+        play_speed=1,
+        title_number=1,
+        title_length=1,
+        title_location=1,
+        chapter_number=1,
+        chapter_length=1,
+        chapter_location=1,
+    )
+    mock_device.dispatcher.send(kaleidescape_const.PLAY_STATUS)
+    await asyncio.sleep(0)
```

Comment:
```suggestion
```

Reply:
These shouldn't be needed since `async_block_till_done` should already sleep 0 at least once

Reply:
Removed. For some reason early in development this was needed. Working find without it now.